src/hotspot/share/oops/oopCast.inline.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
```

Comment:
This is a utility class I've added to be able to do checked casts between `oop` types.